aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/PersistentMutationOutbox.java

```diff
@@ -330,12 +332,27 @@ private Completable handleIncomingUpdate() {
                     return overwriteExistingAndNotify(PendingMutation.Type.CREATE, QueryPredicates.all());
                 case UPDATE:
                     if (QueryPredicates.all().equals(incoming.getPredicate())) {
-                        // If the incoming update does not have a condition, we want to delete any
-                        // existing mutations for the modelId before saving the incoming one.
-                        return removeNotLocking(existing.getMutationId()).andThen(saveIncomingAndNotify());
+                        // If the incoming & existing update is of type serializedModel
+                        // then merge the existing model data into incoming.
+                        if (incoming.getMutatedItem() instanceof SerializedModel
+                                && existing.getMutatedItem() instanceof SerializedModel) {
+                            SerializedModel mergedSerializedModel = SerializedModel.merge(
+                                    (SerializedModel) incoming.getMutatedItem(),
+                                    (SerializedModel) existing.getMutatedItem(),
+                                    incoming.getModelSchema());
+                            PendingMutation<T> mergedPendingMutation = (PendingMutation<T>) PendingMutation.update(
+                                    mergedSerializedModel,
+                                    incoming.getModelSchema());
+                            return removeNotLocking(existing.getMutationId())
+                                    .andThen(saveIncomingAndNotify(mergedPendingMutation));
+                        } else {
+                            // If the incoming update does not have a condition, we want to delete any
```

Comment:
I think this comment makes more sense up above where it was before.  Can you move it back?